message_ix/macro.py

```diff
@@ -388,22 +402,24 @@ def add_model_data(base, clone, data):
     c.read_data()
     c.derive_data()
 
+    # removing old initializeyear_macro before adding new one
+    cat = clone.set("cat_year", {"type_year": "initializeyear_macro"})
+    if not cat.empty:
+        clone.remove_set("cat_year", cat)
+
     # add temporal set structure
     clone.add_set("type_year", "initializeyear_macro")
     clone.add_set("cat_year", ["initializeyear_macro", c.init_year])
-    clone.add_set("type_year", "baseyear_macro")
-    clone.add_set("cat_year", ["baseyear_macro", c.base_year])
 
     # add nodal set structure
     clone.add_set("type_node", "economy")
     for n in c.nodes:
         clone.add_set("cat_node", ["economy", n])
 
     # add sectoral set structure
-    # TODO we shouldn't need a loop here
-    for s in c.sectors:
-        clone.add_set("sector", s)
-        clone.add_set("mapping_macro_sector", [s, s, "useful"])
+    for s, l in product(c.sectors, c.levels):
+        clone.add_set('sector', s)
+        clone.add_set("mapping_macro_sector", [s, s, l])
```

Comment:
Here we assume sector and commodity names are the same. We could differentiate this by using a mapping table obtained from the excel "config" sheet including a commodity column. 

```
    sector_mapping_df = c.sector_mapping

    for i in sector_mapping_df.index:
        s = sector_mapping_df.iloc[i, sector_mapping_df.columns.get_loc("sector")]
        c = sector_mapping_df.iloc[i, sector_mapping_df.columns.get_loc("commodity")]
        l = sector_mapping_df.iloc[i, sector_mapping_df.columns.get_loc("level")]
        clone.add_set('sector', s)
        clone.add_set("mapping_macro_sector", [s, c, l])
```

Reply:
@GamzeUnlu95 , nice suggestion. If possible, please make a PR and add these suggestions in the code, and in the Excel file, so that I can run and approve them.

Reply:
Done in [this PR.](https://github.com/behnam-zakeri/message_ix/pull/9)